File: 03-week-3--additional-practice/00-bonus-1--balancing-parenetheses/ruby/balancing_parentheses.rb
def balancing_parentheses(string)
  openingP = 0
  closingP = 0
  string.chars.each do |char|
    if char == '('
      openingP += 1
      next
    end
    if openingP > 0
      openingP -= 1
    else
      closingP += 1
    end
  end
  openingP + closingP
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts balancing_parentheses('(()())')

  puts

  puts "Expecting: 2"
  puts balancing_parentheses('()))')

  puts

  puts "Expecting: 1"
  puts balancing_parentheses(')')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 03-week-3--additional-practice/00-bonus-1--balancing-parenetheses/javascript/balancing_parentheses.js
function balancingParentheses(string) {
  let openingP = 0
  let closingP = 0
  for (let i = 0; i < string.length; ++i){
    if (string[i] === '('){
      ++openingP
      continue
    }
    if (openingP > 0){
      --openingP
    } else {
      ++closingP
    }
  }
  return openingP + closingP
  
}


if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log(balancingParentheses('(()())'));

  console.log("");

  console.log("Expecting: 2");
  console.log(balancingParentheses('()))'));

  console.log("");

  console.log("Expecting: 1");
  console.log(balancingParentheses(')'));
}

module.exports = balancingParentheses;

// Please add your pseudocode to this file
// And a written explanation of your solution
